pkg/controller/clone-controller.go

```diff
@@ -102,162 +185,150 @@ func (c *CloneController) Run(threadiness int, stopCh <-chan struct{}) error {
+		return nil, nil
+	} else if len(podList) > 1 {
+		return nil, errors.Errorf("multiple pods found for clone PVC %s/%s", pvc.Namespace, pvc.Name)
+	} 
+	return podList, nil
+}
+
 // Create the cloning source and target pods based the pvc. The pvc is checked (again) to ensure that we are not already
 // processing this pvc, which would result in multiple pods for the same pvc.
 func (c *CloneController) processPvcItem(pvc *v1.PersistentVolumeClaim) error {
-	cr, err := getCloneRequestPVC(pvc)
+	anno := map[string]string{}
+	
+	// find import Pod
+	pods, err := c.findClonePodsFromCache(pvc)
 	if err != nil {
 		return err
 	}
-
-	// check our existing pvc one more time to ensure we should be working on it
-	// and to help mitigate any race conditions. This time we get the latest pvc.
-	doCreate, pvc, err := checkClonePVC(c.clientset, pvc, true)
-	if err != nil { // maybe an intermittent api error
+	pvcKey, err := cache.MetaNamespaceKeyFunc(pvc)
+	if err != nil {
 		return err
 	}
-	if !doCreate { // don't create pods but not an error
-		return nil // forget key; logging already done
-	}
 
-	// update pvc with 'AnnCloningPods' annotation to indicate cloning is in process
-	anno := map[string]string{AnnCloningPods: "exist"}
-	var lab map[string]string
-	pvc, err = updatePVC(c.clientset, pvc, anno, lab)
-	if err != nil {
-		return errors.WithMessage(err, "could not update pvc %q annotation and/or label")
+	// Pods must be controlled by this PVC
+	for _, pod := range pods {
+		if pod != nil && !metav1.IsControlledBy(pod, pvc) {
+			return errors.Errorf("found pod %s/%s not owned by pvc %s/%s", pod.Namespace, pod.Name, pvc.Namespace, pvc.Name)
+	 	}
 	}
 
-	//create random string to be used for pod labeling and hostpath name
-	generatedLabelStr := util.RandAlphaNum(GENERATED_CLONING_LABEL_LEN)
-	//create the source pod
-	pod, err := CreateCloneSourcePod(c.clientset, c.cloneImage, c.verbose, c.pullPolicy, cr, pvc, generatedLabelStr)
-	if err != nil {
-		//TODO: remove annotation AnnCloningPods from pvc as pod failed to run
-		return err
+	// expectations prevent us from creating multiple pods. An expectation forces
+	// us to observe a pod's creation in the cache.
+	needsSync := c.podExpectations.SatisfiedExpectations(pvcKey)
+
+	// make sure not to reprocess a PVC that has already completed successfully,
+	// even if the pod no longer exists
+	previousPhase, exists := pvc.ObjectMeta.Annotations[AnnClonePodPhase]
+	if exists && (previousPhase == string(v1.PodSucceeded)) {
+		needsSync = false
 	}
 
-	//create the target pod
-	_, err = CreateCloneTargetPod(c.clientset, c.cloneImage, c.verbose, c.pullPolicy, pvc, generatedLabelStr, pod.ObjectMeta.Namespace)
-	if err != nil {
-		//TODO: remove annotation AnnCloningPods from pvc as pod failed to run
-		return err
+	if pods == nil && needsSync {
+		cr, err := getCloneRequestPVC(pvc)
+		if err != nil {
+			return err
+		}
+
+		// all checks passed, let's create the cloner pods!
+		c.expectPodCreate(pvcKey)
```

Comment:
each pod creation needs an expectation associated with it

Reply:
so How do we distinguish between them? The pvcKey is the same for both pods.

Reply:
the expectation is a ref count for how many pods you expect to be created before allowing the sync to occur again. The key in this case is associated with the PVC not the Pods. 

So, the pvc workload controller is expecting 2 pods to be created. This means incrementing the expectations ref +1 for each pod create, and decrementing -1 for each observed pod create. 

> c.findClonePodsFromCache(pvc) still returns one pod.

both pods need to be returned in this function.

Reply:
Is this how you expect two pods?
func (c *CloneController) expectPodCreate(pvcKey string) {
	c.podExpectations.ExpectCreations(pvcKey, 2)
}

Reply:
call ```c.expectPodCreate``` once right before each create.  Since there are two creations occurring, ```c.expectPodCreate``` will be called twice.